app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt

```diff
@@ -0,0 +1,99 @@
+
+import org.mozilla.fenix.helpers.TestAssetHelper
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Rule
+import org.junit.Before
+import org.junit.After
+import org.junit.Test
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ *  Tests for verifying the main three dot menu options
+ *
+ */
+
+class SettingsAddonsTest {
+    /* ktlint-disable no-blank-line-before-rbrace */ // This imposes unreadable grouping.
+
+    private val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer().apply {
+            setDispatcher(AndroidAssetDispatcher())
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    // Walks through settings add-ons menu to ensure all items are present
+    fun settingsAddonsItemsTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+            verifyAdvancedHeading()
+            verifyAddonsButton()
+        }.openAddonsManagerMenu {
+            verifyAddonsItems()
+        }
+    }
+
+    @Test
+    // Opens a webpage and installs an add-on from the three-dot menu
+    fun installAddonFromThreeDotMenu() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        val addonName = "uBlock Origin"
+
+        navigationToolbar {
+        }.openNewTabAndEnterToBrowser(defaultWebPage.url) {
+            verifyPageContent(defaultWebPage.content)
+        }.openThreeDotMenu {
+        }.openAddonsManagerMenu {
+            clickInstallAddon(addonName)
+            verifyAddonPrompt(addonName)
+            cancelInstallAddon()
+            clickInstallAddon(addonName)
+            acceptInstallAddon()
+            verifyDownloadAddonPrompt()
+        }
+    }
+
+    @Test
+    fun verifyAddonsCanBeUninstalled() {
+        val addonName = "uBlock Origin"
+
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+            verifyAdvancedHeading()
+            verifyAddonsButton()
+        }.openAddonsManagerMenu {
+            clickInstallAddon(addonName)
+            acceptInstallAddon()
+            verifyDownloadAddonPrompt()
+        }.openDetailedMenuForAddon(addonName) {
+            verifyCurrentAddonMenu()
+        }.removeAddon {
+        }
```

Comment:
Do we need any verification here after we remove the add-on? Check for an empty list?

Reply:
Updated with verification, I thought I added it before :)